airbyte-integrations/connectors/source-zendesk-support/source_zendesk_support/streams.py

```diff
@@ -424,9 +425,11 @@ def request_params(self, next_page_token: Mapping[str, Any] = None, **kwargs) ->
         return params
 
 
-class SourceZendeskSupportCursorPaginationStream(SourceZendeskSupportFullRefreshStream):
+class SourceZendeskSupportOffsetPaginationStream(SourceZendeskSupportFullRefreshStream):
     """
-    Endpoints provide a cursor pagination and sorting mechanism
+    Functionality of this class copied into a SourceZendeskSupportPaginationStream class and modified to suit rudderstack needs.
```

Comment:
Comment `Functionality of this class copied into a SourceZendeskSupportPaginationStream class and modified to suit rudderstack needs.` seems a bit misplaced. Copied to `SourceZendeskSupportCursorPaginationStream` you meant?

Also instead of this comment, should we support OffsetPaginationStream class and CursorPaginationStream class explicitly and use both wherever needed? As in streams not supporting Cursor based will need offsetPagination? If all streams have support for cursor based, we can plan to get rid of this call.

Reply:
Missed updating comments after I changed class names. Will update.

> should we support OffsetPaginationStream class and CursorPaginationStream class explicitly and use both wherever needed

This would be more descriptive actually. Originally was planning to remove `OffsetPaginationStream` so didn't do this. But since we have this class now it makes more sense to do as you suggested. Will change